pkg/cvo/upgradeable.go

```diff
@@ -0,0 +1,215 @@
+package cvo
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog"
+
+	"github.com/openshift/cluster-version-operator/lib/resourcemerge"
+)
+
+// syncUpgradeable. The status is only checked if it has been more than
+// the minimumUpdateCheckInterval since the last check.
+func (optr *Operator) syncUpgradeable(config *configv1.ClusterVersion) error {
+	// updates are only checked at most once per minimumUpdateCheckInterval or if the generation changes
+	u := optr.getUpgradeable()
+	if u != nil && u.RecentlyChanged(optr.minimumUpdateCheckInterval) {
+		klog.V(4).Infof("Upgradeable conditions were recently checked, will try later.")
+		return nil
+	}
+
+	now := metav1.Now()
+	var conds []configv1.ClusterOperatorStatusCondition
+	var reasons []string
+	for _, check := range optr.upgradeableChecks {
+		if cond := check.Check(); cond != nil {
+			reasons = append(reasons, cond.Reason)
+			cond.LastTransitionTime = now
+			conds = append(conds, *cond)
+		}
+	}
+	if len(conds) == 1 {
+		conds = append(conds, configv1.ClusterOperatorStatusCondition{
+			Type:               configv1.OperatorUpgradeable,
+			Status:             configv1.ConditionFalse,
+			Reason:             conds[0].Reason,
+			Message:            conds[0].Message,
+			LastTransitionTime: now,
+		})
+	} else if len(conds) > 1 {
+		conds = append(conds, configv1.ClusterOperatorStatusCondition{
+			Type:               configv1.OperatorUpgradeable,
+			Status:             configv1.ConditionFalse,
+			Reason:             "MultipleReasons",
+			Message:            fmt.Sprintf("Cluster is not Upgradeable due to multiple reasons: %s", strings.Join(reasons, ",")),
```

Comment:
Should be `Cluster is not upgradeable ...`.  Don't capitalize the condition.

Reply:
fixed with https://github.com/openshift/cluster-version-operator/pull/243/commits/03e188b3d7fcf3e2e374687186d8c994685d605f